shared/functions.ts

```diff
@@ -4,14 +4,17 @@ import multihash from 'multihashes'
 import nacl from 'tweetnacl'
 import blake from 'blakejs'
 
-// Makes the `Buffer` global available in the browser if needed.
-if (typeof window === 'object' && typeof Buffer === 'undefined') {
-  // Only import `Buffer` to hopefully help treeshaking.
-  const { Buffer } = require('buffer')
+import { Buffer } from 'buffer'
+
+if (typeof window === 'object') {
```

Comment:
Do you mean just using Deno's `Buffer` instead of the third-party `buffer` module? That would be nice, but the frontend is also using this file, and there is no browser `Buffer` API yet. :/

Reply:
I assumed (but didn't test) that `globalThis` and `self` are the same (and `widnow` and `self` are the same), and hence the `if` is not needed by using `self`. Re. browser `Buffer` support, it's unlikely to ever happen because it's non-standard. The standard way is `Uint8Array` (`Buffer` is this plus some helpers for encoding and decoding). It might be worth it to move away from `Buffer` later but probably not relevant to this PR.

Reply:
You were right, thank you!